File: leetcode/easy/242_有效的字母异位词/242_有效的字母异位词.go
//哈希表法
func isAnagram(s string, t string) bool {
    record := [26]int{}
    for _, char := range s {
        record[char-'a']++
    }
    for _, char := range t {
        record[char-'a']--
    }
    for i := 0; i < 26; i++ {
        if record[i] != 0 {
            return false
        }
    }
    return true
}

File: leetcode/easy/242_有效的字母异位词/242_有效的字母异位词.py
# 哈希表法
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        record = [0] * 26
        for char in s:
            record[ord(char) - ord('a')] += 1
        for char in t:
            record[ord(char) - ord('a')] -= 1
        for count in record:
            if count != 0:
                return False
        return True



#使用Counter
from collections import Counter
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        return Counter(s) == Counter(t)